French tts dailun

lidl2019 opened the pull request:
Please ensure you've followed the checklist and provide all the required information *before* requesting a review.

## Required Information

- [x] Reference the issue this is meant to fix or describe it if none exists.
- [x] Full description of changes made.

## Pull Request Checklist

- [x] Coding standards are followed (e.g., [PEP8](https://pep8.org/))
- [x] An entry exists for the container in `docker-compose.yml` or `build.yml`
- [x] The Docker image builds
- [x] The container runs correctly when sent inputs with the changes
- [x] No models or other large files are part of these commits
- [x] A description of the tests already run as part of this PR is present
- [x] CI is set up under `.github/workflows` or is not applicable

lidl2019 committed:
TODO vocoder in unsupported format

lidl2019 committed:
Vocoder not in correct format

lidl2019 committed:
added tts deployment

lidl2019 commented:
fixed bugs

JRegimbal commented on services/espnet-tts/segment.request.json:
```diff
@@ -0,0 +1,27 @@
+{
```
This file should not be here.

lidl2019 replied:
After I shifted my code to use absolute paths, when I execute "docker build -t tts ." in the "/services/espnet-tts"directory 
The following error raised:
Step 10/25 : COPY /services/espnet-tts/requirements.txt .
COPY failed: file not found in build context or excluded by .dockerignore: stat services/espnet-tts/requirements.txt: file does not exis

JRegimbal commented on services/espnet-tts/segment.response.json:
```diff
@@ -0,0 +1,33 @@
+{
```
This file should not be here.

JRegimbal commented on services/espnet-tts/src/espnet_util.py:
```diff
@@ -23,46 +26,78 @@
     logger.info(f"Elapsed text2speech: {t2 - start}")
     logger.info(f"Elapsed vocoder: {t3 - t2}")
+
+    # audio = Audio(wav.view(-1).cpu().numpy(), rate=text2speech.fs)
```
Don't add commented code in here!

JRegimbal commented on services/espnet-tts/src/espnet_util.py:
```diff
@@ -23,46 +26,78 @@
-    speed_control_alpha=1.0,
+
+
+# text2speech = Text2Speech(
```
Delete!

JRegimbal commented on services/espnet-tts/Dockerfile:
```diff
@@ -1,7 +1,9 @@
+FROM pytorch/pytorch:1.12.1-cuda11.3-cudnn8-runtime
 
 RUN apt-get update && apt-get install -y libsndfile1 build-essential && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && apt-get install -y espeak-ng
```
Include in installs above so we clear lists after

JRegimbal commented on services/espnet-tts/Dockerfile:
```diff
@@ -10,18 +12,43 @@ USER python
 RUN pip install --upgrade pip
-COPY /services/espnet-tts/requirements.txt .
+COPY ./requirements.txt .
+# COPY /services/espnet-tts/requirements.txt .
```
You cannot actually use relative paths here for how the build actually works. In order to access the schemas, your working directory for builds **must** be the root of the repository. This is why you needed to copy the schema files below (needlessly).

JRegimbal reviewed (changes requested):
This will not work. Your entire context is shifted to the local directory, it **must** be run from the root of the repository. This is why absolute paths are used. Add an entry to `build.yml` for this. Contact me if you have more questions.

lidl2019 committed:
update to absolute path

lidl2019 committed:
remove unnecessary files

JRegimbal committed:
Restore improperly deleted files

These need to be kept since we aren't replacing English TTS. They should
have been copied and modified, not modified in place and moved.

JRegimbal committed:
Add workflow file for ESPNet French TTS Service

JRegimbal commented:
@lidl2019 Please note that the code as of now does not pass PEP 8 linting. Check the details of the failed GitHub Action for more information.

JRegimbal reviewed (commented):
Not fully cleaned up (see other comments). I've gone and added the workflow file so you get actions for this and undid your changes in the normal `espnet-tts` directory since that is not supposed to be modified as part of this PR.

lidl2019 committed:
solve for lint check

lidl2019 committed:
style checked for espnet_utils.py

JRegimbal committed:
Fix service name

JRegimbal committed:
Remove unnecessary files, update README

JRegimbal reviewed (approved):
Great job, Dailun! Both endpoints work as expected. As you work on your final report, please add a brief summary of how you trained the tacotron model in `services/espnet-tts-fr/README.md`.
